{lib}[foss/2021a] DeepLabCut v2.2.0.6 w/ Python 3.9.5

sassy-crick opened the pull request:
(created using `eb --new-pr`)

Requires:
- [x] PR #16282
- [x] PR #16479

jfgrimm commented on easybuild/easyconfigs/d/DeepLabCut/DeepLabCut-2.2.0.6-foss-2021a-CUDA-11.3.1.eb:
```diff
@@ -0,0 +1,68 @@
+toolchainopts = {'pic': True}
+
+dependencies = [
+    ('CUDA', '11.3.1', '', True),
+    ('cuDNN', '8.2.1.32', '-CUDA-%(cudaver)s', True),
```
```suggestion
    ('CUDA', '11.3.1', '', SYSTEM),
    ('cuDNN', '8.2.1.32', '-CUDA-%(cudaver)s', SYSTEM),
```

jfgrimm replied:
@sassy-crick one of your commits has reverted back to `True`

jfgrimm commented on easybuild/easyconfigs/d/DeepLabCut/DeepLabCut-2.2.0.6-foss-2021a-CUDA-11.3.1.eb:
```diff
@@ -0,0 +1,68 @@
+        'sources': ['%(name)s-%(version)s.zip'],
+        'checksums': ['4f2a4d39e4ea601b9ab42b2db08b5918a9538c168cff1c6895ae26646f3d73b1'],
+    }),
+    ('pandas', '1.3.3', {
```
we bundle `pandas` in `SciPy-bundle`, is there a need for a version newer than `1.2.4`?

sassy-crick replied:
Probably no need, most likely I did not know and I was thus looking for the latest version of it. I have removed it and it is still building. Thanks for pointing that out.

casparvl commented:
Test report by @casparvl
**SUCCESS**
Build succeeded for 0 out of 0 (1 easyconfigs in total)
gcn2 - Linux RHEL 8.4, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz, 4 x NVIDIA NVIDIA A100-SXM4-40GB, 515.43.04, Python 3.6.8
See https://gist.github.com/cf299f518231c74ce555218b2ab1c3aa for a full test report.

casparvl commented:
Test report by @casparvl
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
gcn2 - Linux RHEL 8.4, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz, 4 x NVIDIA NVIDIA A100-SXM4-40GB, 515.43.04, Python 3.6.8
See https://gist.github.com/e36fe7607c8a9a0e3d74642be011e7c0 for a full test report.

casparvl commented:
Test report by @casparvl
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
software2.lisa.surfsara.nl - Linux debian 10.13, x86_64, Intel(R) Xeon(R) Bronze 3104 CPU @ 1.70GHz, 4 x NVIDIA NVIDIA TITAN V, 470.103.01, Python 3.7.3
See https://gist.github.com/5d7d379e53a7628a58748e17b8b49f61 for a full test report.

casparvl commented:
@boegelbot please test @ generoso

casparvl commented:
Test report by @casparvl
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
software2.lisa.surfsara.nl - Linux debian 10.13, x86_64, Intel(R) Xeon(R) Bronze 3104 CPU @ 1.70GHz, 4 x NVIDIA NVIDIA TITAN V, 470.103.01, Python 3.7.3
See https://gist.github.com/39fcce90b63911270131bbeb014085f7 for a full test report.

casparvl commented:
Test report by @casparvl
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
gcn2.local.snellius.surf.nl - Linux RHEL 8.4, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz, 4 x NVIDIA NVIDIA A100-SXM4-40GB, 515.43.04, Python 3.6.8
See https://gist.github.com/9c4671f9f308b0c35c5096b8d4d8626e for a full test report.

casparvl commented:
From the pip sanity check:
```
== 2022-10-17 16:01:53,979 build_log.py:169 ERROR EasyBuild crashed with an error (at easybuild/RHEL8/2022/software/EasyBuild/4.6.1/lib/python3.6/site-packages/easybuild/base/exceptions.py:124 in __init__): `pip check` failed:
deeplabcut 2.2.0.6 requires moviepy, which is not installed.
deeplabcut 2.2.0.6 has requirement scikit-image<=0.18.1,>=0.17, but you have scikit-image 0.18.3.
```

sassy-crick commented:
> From the pip sanity check:
> 
> ```
> == 2022-10-17 16:01:53,979 build_log.py:169 ERROR EasyBuild crashed with an error (at easybuild/RHEL8/2022/software/EasyBuild/4.6.1/lib/python3.6/site-packages/easybuild/base/exceptions.py:124 in __init__): `pip check` failed:
> deeplabcut 2.2.0.6 requires moviepy, which is not installed.
> deeplabcut 2.2.0.6 has requirement scikit-image<=0.18.1,>=0.17, but you have scikit-image 0.18.3.
> ```

Thanks for finding this, entirely my wrong as I did not rebuild after I added the `sanity_pip_check = True`. I have added the missing `MoviePy` and also downgraded `scikit-image`. To avoid (hopefully) any conflicts, I have installed `scikit-image` as an extension, rather than as a module. If that is the wrong way forward, please let me know. I was hoping that extensions might be the better way here. 
I have also updated PR #16281 regarding this.

jfgrimm commented:
@sassy-crick take a look at: https://github.com/DeepLabCut/DeepLabCut/issues/1791, https://github.com/DeepLabCut/DeepLabCut/pull/1800

we can probably get away with just using the `scikit-image` version we have in an easyconfig, and loosening the upper version bound in `requirements.txt`

sassy-crick commented:
> @sassy-crick take a look at: [DeepLabCut/DeepLabCut#1791](https://github.com/DeepLabCut/DeepLabCut/issues/1791), [DeepLabCut/DeepLabCut#1800](https://github.com/DeepLabCut/DeepLabCut/pull/1800)
> 
> we can probably get away with just using the `scikit-image` version we have in an easyconfig, and loosening the upper version bound in `requirements.txt`

Done. I am always a bit cautious to changing upper bounds but it builds ok.

I have also reverted the `True` back to `System`, which happened accidentally.

casparvl commented:
Test report by @casparvl
**SUCCESS**
Build succeeded for 2 out of 2 (1 easyconfigs in total)
gcn1.local.snellius.surf.nl - Linux RHEL 8.4, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz, 4 x NVIDIA NVIDIA A100-SXM4-40GB, 515.43.04, Python 3.6.8
See https://gist.github.com/a4115ec75e873ccafd53bc72c63f472a for a full test report.

casparvl reviewed (approved):
Looks good to me!

casparvl commented:
Test report by @casparvl
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
software2.lisa.surfsara.nl - Linux debian 10.13, x86_64, Intel(R) Xeon(R) Bronze 3104 CPU @ 1.70GHz, 4 x NVIDIA NVIDIA TITAN V, 470.103.01, Python 3.7.3
See https://gist.github.com/c26d821f54b769ce6bab582bce553b5f for a full test report.
